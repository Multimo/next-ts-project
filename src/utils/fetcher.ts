// isomophic fetch since fetch only works only exists on the window
import fetch from 'node-fetch';


export const fetcher = async (url: string) => {
  try {
    const res = await fetch(`https://jsonplaceholder.typicode.com/${url}`);
    const json = await res.json();
    return json;
  } catch (e) {
    console.log(e);
  }
}

export const getAlbums = () => fetcher('albums');
export const getAlbumPhotos = (id: string) => fetcher(`albums/${id}/photos`);
export const getAlbumById = (id: string) => fetcher(`albums/${id}`);

export const getPosts = () => fetcher('posts');
export const getPostComments = (id: string) => fetcher(`posts/${id}/comments`);
export const getPostById = (id: string) => fetcher(`posts/${id}`);