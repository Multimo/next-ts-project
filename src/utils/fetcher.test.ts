import { fetcher } from "./fetcher";
jest.mock('../../__mocks__/fetch');

describe('fetchter tests', () => {
  it('should return a function', () => {
    return expect(fetcher('albums')).resolves.toMatchSnapshot();
  });
});