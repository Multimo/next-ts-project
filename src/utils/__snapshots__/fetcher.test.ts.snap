// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fetchter tests should return a function 1`] = `
Array [
  Object {
    "id": 1,
    "title": "quidem molestiae enim",
    "userId": 1,
  },
  Object {
    "id": 2,
    "title": "sunt qui excepturi placeat culpa",
    "userId": 1,
  },
  Object {
    "id": 3,
    "title": "omnis laborum odio",
    "userId": 1,
  },
  Object {
    "id": 4,
    "title": "non esse culpa molestiae omnis sed optio",
    "userId": 1,
  },
  Object {
    "id": 5,
    "title": "eaque aut omnis a",
    "userId": 1,
  },
  Object {
    "id": 6,
    "title": "natus impedit quibusdam illo est",
    "userId": 1,
  },
  Object {
    "id": 7,
    "title": "quibusdam autem aliquid et et quia",
    "userId": 1,
  },
  Object {
    "id": 8,
    "title": "qui fuga est a eum",
    "userId": 1,
  },
  Object {
    "id": 9,
    "title": "saepe unde necessitatibus rem",
    "userId": 1,
  },
  Object {
    "id": 10,
    "title": "distinctio laborum qui",
    "userId": 1,
  },
  Object {
    "id": 11,
    "title": "quam nostrum impedit mollitia quod et dolor",
    "userId": 2,
  },
  Object {
    "id": 12,
    "title": "consequatur autem doloribus natus consectetur",
    "userId": 2,
  },
  Object {
    "id": 13,
    "title": "ab rerum non rerum consequatur ut ea unde",
    "userId": 2,
  },
  Object {
    "id": 14,
    "title": "ducimus molestias eos animi atque nihil",
    "userId": 2,
  },
  Object {
    "id": 15,
    "title": "ut pariatur rerum ipsum natus repellendus praesentium",
    "userId": 2,
  },
  Object {
    "id": 16,
    "title": "voluptatem aut maxime inventore autem magnam atque repellat",
    "userId": 2,
  },
  Object {
    "id": 17,
    "title": "aut minima voluptatem ut velit",
    "userId": 2,
  },
  Object {
    "id": 18,
    "title": "nesciunt quia et doloremque",
    "userId": 2,
  },
  Object {
    "id": 19,
    "title": "velit pariatur quaerat similique libero omnis quia",
    "userId": 2,
  },
  Object {
    "id": 20,
    "title": "voluptas rerum iure ut enim",
    "userId": 2,
  },
  Object {
    "id": 21,
    "title": "repudiandae voluptatem optio est consequatur rem in temporibus et",
    "userId": 3,
  },
  Object {
    "id": 22,
    "title": "et rem non provident vel ut",
    "userId": 3,
  },
  Object {
    "id": 23,
    "title": "incidunt quisquam hic adipisci sequi",
    "userId": 3,
  },
  Object {
    "id": 24,
    "title": "dolores ut et facere placeat",
    "userId": 3,
  },
  Object {
    "id": 25,
    "title": "vero maxime id possimus sunt neque et consequatur",
    "userId": 3,
  },
  Object {
    "id": 26,
    "title": "quibusdam saepe ipsa vel harum",
    "userId": 3,
  },
  Object {
    "id": 27,
    "title": "id non nostrum expedita",
    "userId": 3,
  },
  Object {
    "id": 28,
    "title": "omnis neque exercitationem sed dolor atque maxime aut cum",
    "userId": 3,
  },
  Object {
    "id": 29,
    "title": "inventore ut quasi magnam itaque est fugit",
    "userId": 3,
  },
  Object {
    "id": 30,
    "title": "tempora assumenda et similique odit distinctio error",
    "userId": 3,
  },
  Object {
    "id": 31,
    "title": "adipisci laborum fuga laboriosam",
    "userId": 4,
  },
  Object {
    "id": 32,
    "title": "reiciendis dolores a ut qui debitis non quo labore",
    "userId": 4,
  },
  Object {
    "id": 33,
    "title": "iste eos nostrum",
    "userId": 4,
  },
  Object {
    "id": 34,
    "title": "cumque voluptatibus rerum architecto blanditiis",
    "userId": 4,
  },
  Object {
    "id": 35,
    "title": "et impedit nisi quae magni necessitatibus sed aut pariatur",
    "userId": 4,
  },
  Object {
    "id": 36,
    "title": "nihil cupiditate voluptate neque",
    "userId": 4,
  },
  Object {
    "id": 37,
    "title": "est placeat dicta ut nisi rerum iste",
    "userId": 4,
  },
  Object {
    "id": 38,
    "title": "unde a sequi id",
    "userId": 4,
  },
  Object {
    "id": 39,
    "title": "ratione porro illum labore eum aperiam sed",
    "userId": 4,
  },
  Object {
    "id": 40,
    "title": "voluptas neque et sint aut quo odit",
    "userId": 4,
  },
  Object {
    "id": 41,
    "title": "ea voluptates maiores eos accusantium officiis tempore mollitia consequatur",
    "userId": 5,
  },
  Object {
    "id": 42,
    "title": "tenetur explicabo ea",
    "userId": 5,
  },
  Object {
    "id": 43,
    "title": "aperiam doloremque nihil",
    "userId": 5,
  },
  Object {
    "id": 44,
    "title": "sapiente cum numquam officia consequatur vel natus quos suscipit",
    "userId": 5,
  },
  Object {
    "id": 45,
    "title": "tenetur quos ea unde est enim corrupti qui",
    "userId": 5,
  },
  Object {
    "id": 46,
    "title": "molestiae voluptate non",
    "userId": 5,
  },
  Object {
    "id": 47,
    "title": "temporibus molestiae aut",
    "userId": 5,
  },
  Object {
    "id": 48,
    "title": "modi consequatur culpa aut quam soluta alias perspiciatis laudantium",
    "userId": 5,
  },
  Object {
    "id": 49,
    "title": "ut aut vero repudiandae voluptas ullam voluptas at consequatur",
    "userId": 5,
  },
  Object {
    "id": 50,
    "title": "sed qui sed quas sit ducimus dolor",
    "userId": 5,
  },
  Object {
    "id": 51,
    "title": "odit laboriosam sint quia cupiditate animi quis",
    "userId": 6,
  },
  Object {
    "id": 52,
    "title": "necessitatibus quas et sunt at voluptatem",
    "userId": 6,
  },
  Object {
    "id": 53,
    "title": "est vel sequi voluptatem nemo quam molestiae modi enim",
    "userId": 6,
  },
  Object {
    "id": 54,
    "title": "aut non illo amet perferendis",
    "userId": 6,
  },
  Object {
    "id": 55,
    "title": "qui culpa itaque omnis in nesciunt architecto error",
    "userId": 6,
  },
  Object {
    "id": 56,
    "title": "omnis qui maiores tempora officiis omnis rerum sed repellat",
    "userId": 6,
  },
  Object {
    "id": 57,
    "title": "libero excepturi voluptatem est architecto quae voluptatum officia tempora",
    "userId": 6,
  },
  Object {
    "id": 58,
    "title": "nulla illo consequatur aspernatur veritatis aut error delectus et",
    "userId": 6,
  },
  Object {
    "id": 59,
    "title": "eligendi similique provident nihil",
    "userId": 6,
  },
  Object {
    "id": 60,
    "title": "omnis mollitia sunt aliquid eum consequatur fugit minus laudantium",
    "userId": 6,
  },
  Object {
    "id": 61,
    "title": "delectus iusto et",
    "userId": 7,
  },
  Object {
    "id": 62,
    "title": "eos ea non recusandae iste ut quasi",
    "userId": 7,
  },
  Object {
    "id": 63,
    "title": "velit est quam",
    "userId": 7,
  },
  Object {
    "id": 64,
    "title": "autem voluptatem amet iure quae",
    "userId": 7,
  },
  Object {
    "id": 65,
    "title": "voluptates delectus iure iste qui",
    "userId": 7,
  },
  Object {
    "id": 66,
    "title": "velit sed quia dolor dolores delectus",
    "userId": 7,
  },
  Object {
    "id": 67,
    "title": "ad voluptas nostrum et nihil",
    "userId": 7,
  },
  Object {
    "id": 68,
    "title": "qui quasi nihil aut voluptatum sit dolore minima",
    "userId": 7,
  },
  Object {
    "id": 69,
    "title": "qui aut est",
    "userId": 7,
  },
  Object {
    "id": 70,
    "title": "et deleniti unde",
    "userId": 7,
  },
  Object {
    "id": 71,
    "title": "et vel corporis",
    "userId": 8,
  },
  Object {
    "id": 72,
    "title": "unde exercitationem ut",
    "userId": 8,
  },
  Object {
    "id": 73,
    "title": "quos omnis officia",
    "userId": 8,
  },
  Object {
    "id": 74,
    "title": "quia est eius vitae dolor",
    "userId": 8,
  },
  Object {
    "id": 75,
    "title": "aut quia expedita non",
    "userId": 8,
  },
  Object {
    "id": 76,
    "title": "dolorem magnam facere itaque ut reprehenderit tenetur corrupti",
    "userId": 8,
  },
  Object {
    "id": 77,
    "title": "cupiditate sapiente maiores iusto ducimus cum excepturi veritatis quia",
    "userId": 8,
  },
  Object {
    "id": 78,
    "title": "est minima eius possimus ea ratione velit et",
    "userId": 8,
  },
  Object {
    "id": 79,
    "title": "ipsa quae voluptas natus ut suscipit soluta quia quidem",
    "userId": 8,
  },
  Object {
    "id": 80,
    "title": "id nihil reprehenderit",
    "userId": 8,
  },
  Object {
    "id": 81,
    "title": "quibusdam sapiente et",
    "userId": 9,
  },
  Object {
    "id": 82,
    "title": "recusandae consequatur vel amet unde",
    "userId": 9,
  },
  Object {
    "id": 83,
    "title": "aperiam odio fugiat",
    "userId": 9,
  },
  Object {
    "id": 84,
    "title": "est et at eos expedita",
    "userId": 9,
  },
  Object {
    "id": 85,
    "title": "qui voluptatem consequatur aut ab quis temporibus praesentium",
    "userId": 9,
  },
  Object {
    "id": 86,
    "title": "eligendi mollitia alias aspernatur vel ut iusto",
    "userId": 9,
  },
  Object {
    "id": 87,
    "title": "aut aut architecto",
    "userId": 9,
  },
  Object {
    "id": 88,
    "title": "quas perspiciatis optio",
    "userId": 9,
  },
  Object {
    "id": 89,
    "title": "sit optio id voluptatem est eum et",
    "userId": 9,
  },
  Object {
    "id": 90,
    "title": "est vel dignissimos",
    "userId": 9,
  },
  Object {
    "id": 91,
    "title": "repellendus praesentium debitis officiis",
    "userId": 10,
  },
  Object {
    "id": 92,
    "title": "incidunt et et eligendi assumenda soluta quia recusandae",
    "userId": 10,
  },
  Object {
    "id": 93,
    "title": "nisi qui dolores perspiciatis",
    "userId": 10,
  },
  Object {
    "id": 94,
    "title": "quisquam a dolores et earum vitae",
    "userId": 10,
  },
  Object {
    "id": 95,
    "title": "consectetur vel rerum qui aperiam modi eos aspernatur ipsa",
    "userId": 10,
  },
  Object {
    "id": 96,
    "title": "unde et ut molestiae est molestias voluptatem sint",
    "userId": 10,
  },
  Object {
    "id": 97,
    "title": "est quod aut",
    "userId": 10,
  },
  Object {
    "id": 98,
    "title": "omnis quia possimus nesciunt deleniti assumenda sed autem",
    "userId": 10,
  },
  Object {
    "id": 99,
    "title": "consectetur ut id impedit dolores sit ad ex aut",
    "userId": 10,
  },
  Object {
    "id": 100,
    "title": "enim repellat iste",
    "userId": 10,
  },
]
`;
